pinot-query-planner/src/main/java/org/apache/pinot/calcite/rel/rules/PinotAggregateReduceFunctionsRule.java

```diff
@@ -16,21 +16,27 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-package org.apache.pinot.calcite.sql.fun;
+package org.apache.pinot.calcite.rel.rules;
 
-import org.apache.calcite.sql.SqlCall;
-import org.apache.calcite.sql.SqlNode;
-import org.apache.calcite.sql.fun.SqlCoalesceFunction;
-import org.apache.calcite.sql.validate.SqlValidator;
+import org.apache.calcite.rel.core.AggregateCall;
+import org.apache.calcite.rel.rules.AggregateReduceFunctionsRule;
+import org.apache.calcite.sql.SqlKind;
 
 
 /**
- * Pinot supports native COALESCE function, thus no need to create CASE WHEN conversion.
+ * Pinot customized version of {@link AggregateReduceFunctionsRule} which only reduce on SUM and AVG.
  */
-public class PinotSqlCoalesceFunction extends SqlCoalesceFunction {
+public class PinotAggregateReduceFunctionsRule extends AggregateReduceFunctionsRule {
+  public static final PinotAggregateReduceFunctionsRule INSTANCE =
+      new PinotAggregateReduceFunctionsRule(Config.DEFAULT);
+
+  private PinotAggregateReduceFunctionsRule(Config config) {
+    super(config);
+  }
 
   @Override
-  public SqlNode rewriteCall(SqlValidator validator, SqlCall call) {
-    return call;
+  public boolean canReduce(AggregateCall call) {
+    SqlKind kind = call.getAggregation().getKind();
+    return kind == SqlKind.SUM || kind == SqlKind.AVG;
```

Comment:
Added some javadoc to make it clear. Currently `STDDEV_POP, STDDEV_SAMP, VAR_POP, VAR_SAMP, COVAR_POP, COVAR_SAMP` breaks because of the original rule. Take a look at the changes in `StatisticAggregates.json`

Reply:
Ah okay, makes sense now, thanks!

Reply:
Does this rule applies in the leaf stage or also in the intermediate stage? How we also merge data from different workers in the not simpler form?

Reply:
When this rule is applied, we don't have leaf stage concept yet (leaf stage is determined with `PinotAggregateExchangeNodeInsertRule`).
We don't really need these rewrite because Pinot can directly handle `SUM` and `AVG` with proper null handling (the rule is needed for engine without proper null handling). I didn't directly remove the rule because that is out of the scope of this PR, and null handling support requires some more tweaks.